src/snowflake/snowpark/mock/snowflake_data_type.py

```diff
@@ -339,57 +348,59 @@ def __bool__(self):
 
     def __and__(self, other):
         result = super().__and__(other)
-        result.sf_type = ColumnType(BooleanType(), self.sf_type.nullable)
+        result.sf_type = ColumnType(BooleanType(), True)
         return result
 
     def __or__(self, other):
         result = super().__or__(other)
-        result.sf_type = ColumnType(BooleanType(), self.sf_type.nullable)
+        result.sf_type = ColumnType(BooleanType(), True)
         return result
 
     def __ne__(self, other):
         result = super().__ne__(other)
-        result.sf_type = ColumnType(BooleanType(), self.sf_type.nullable)
+        result.sf_type = ColumnType(BooleanType(), True)
         return result
 
     def __xor__(self, other):
         result = super().__xor__(other)
-        result.sf_type = ColumnType(BooleanType(), self.sf_type.nullable)
+        result.sf_type = ColumnType(BooleanType(), True)
         return result
 
     def __pow__(self, power):
         result = super().__pow__(power)
-        result.sf_type = ColumnType(DoubleType(), self.sf_type.nullable)
+        result.sf_type = ColumnType(
+            FloatType(), self.sf_type.nullable or power.sf_type.nullable
```

Comment:
are all the `FloatType` in the code usage ultimately be converted to `DoubleType` via `normalize_output_sf_type` at some point?
and from user perspective will they get a chance to see the `FloatType` at all?

it seems the two types are interchangeable and we want `DoubleType` to be in the final output. can we just use DoubleType everywhere?